index.js

```diff
@@ -5,7 +5,16 @@ var isglob = require('is-glob');
 var pathDirname = require('path-dirname');
 
 module.exports = function globParent(str) {
-	str += 'a'; // preserves full path in case of trailing path separator
+	// replace braces/brackets sections with *
+	str = str.replace(/(^|[^\\])(\{([^{}]*?)}|\[([^\[\]]*?)\])/g, '$1*');
+	if (/[\}\}]$/.test(str)) str += '/';
```

Comment:
This line is specifically meant to fix https://github.com/es128/glob-parent/pull/11/files#diff-1dd241c4cd3fd1dd89c570cee98b79ddR78

Where the special chars were escaped, but then still being split by dirname